packages/mui-system/src/createStyled/createStyled.js

```diff
@@ -13,81 +17,80 @@ export function shouldForwardProp(prop) {
   return prop !== 'ownerState' && prop !== 'theme' && prop !== 'sx' && prop !== 'as';
 }
 
-function resolveTheme(themeId, theme, defaultTheme) {
-  return isObjectEmpty(theme) ? defaultTheme : theme[themeId] || theme;
-}
-
-const PROCESSED_PROPS = Symbol('mui.processed_props');
-
-function attachTheme(props, themeId, defaultTheme) {
-  if (PROCESSED_PROPS in props) {
-    return props[PROCESSED_PROPS];
-  }
-
-  const processedProps = {
-    ...props,
-    theme: resolveTheme(themeId, props.theme, defaultTheme),
-  };
-
-  props[PROCESSED_PROPS] = processedProps;
-  processedProps[PROCESSED_PROPS] = processedProps;
-
-  return processedProps;
-}
-
 function defaultOverridesResolver(slot) {
   if (!slot) {
     return null;
   }
   return (_props, styles) => styles[slot];
 }
 
-function processStyle(style, props) {
+function attachTheme(props, themeId, defaultTheme) {
+  props.theme = isObjectEmpty(props.theme) ? defaultTheme : props.theme[themeId] || props.theme;
+}
+
+function processStyle(props, style) {
```

Comment:
What was the motivation for changing the order of the args?

Reply:
There's `processStyle` and `processStyleVariants`, both takes `props` and then have diverging arguments, it felt more consistent to start each function with the convergent argument, then the rest.